src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs

```diff
@@ -297,6 +306,14 @@ public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder,
             imageAnnotation.Tag = parsedReference.Tag ?? tag ?? "latest";
         }
 
+        // If there's a DockerfileBuildAnnotation with an image name/tag, clear them
```

Comment:
Why do we clear here but not in `WithImageTag`?

Reply:
I'm tightening this up a bit as I don't think this is all necessary actually.